modules/neovim/neovim.nix

```
@@ -20,14 +21,49 @@
       signColumn = lib.mkEnableOption "background transparency for the Neovim sign column";
       numberLine = lib.mkEnableOption "background transparency for the NeoVim number/relativenumber column";
     };
+
+    pluginColorConfig = lib.mkOption {
+      type = lib.types.lines;
+      default = "";
+      internal = true;
+    };
   };
 
-  config =
-    lib.mkIf (config.stylix.enable && config.stylix.targets.neovim.enable)
+  configElements = [
+    (
+      { colors, cfg }:
+      {
+        stylix.targets.neovim.pluginColorConfig =
```

Unsure how I never thought of this before, but should this not be:

```suggestion
      { colors, cfg }:
      {
        ${cfg}.pluginColorConfig =
```

In that case, this improvement could be applied treewide to existing `mkTarget` modules containing this anti-pattern. The treewide change is obviously out of the scope of this PR.

---

the contents of `config` as supplied by the function argument `cfg` (or in the arguments of any module) are only for getting, not setting. This pattern attempts to subvert the module system and causes an error:
<details>
<summary>Error</summary>

```
error:
       … while checking flake output 'checks'

       … while checking the derivation 'checks.x86_64-linux.doc'

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:37:12:
           36|
           37|   strict = derivationStrict drvAttrs;
             |            ^
           38|

       … while evaluating the option `_module.freeformType':

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: attribute 'stylix' missing
       at /nix/store/zf9zpfzxkjbqiyx9yv4qbgmgnpbzr8vj-source/stylix/mk-target.nix:194:13:
          193|     let
          194|       cfg = config.stylix.targets.${name};
             |             ^
          195|
```
<details>

---

> Unsure how I never thought of this before, but should this not be:

> The contents of `config` as supplied by the function argument `cfg` (or in the arguments of any module) are only for getting, not setting. This pattern attempts to subvert the module system and causes an error:

Now, I know why I never thought of that... Either way, it would be nice to add something similar to `mkTarget` to avoid hard-coding the generated `stylix.targets.${target}` namespace.